airbyte-integrations/connectors/source-gcs/source_gcs/cursor.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from datetime import datetime
+
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.sources.file_based.stream.cursor import DefaultFileBasedCursor
+
+
+class Cursor(DefaultFileBasedCursor):
+    @staticmethod
+    def get_file_uri(file: RemoteFile) -> str:
```

Comment:
I'm a little concerned about trying to keep this in sync with `default_file_based_cursor.py`.

When `RemoteFile`s are created [here](https://github.com/airbytehq/airbyte/pull/31212/files#diff-3e292c0e549de4bfabc96968d6ac8228b0dd511692dde5dd6201b35319b44ba7R83), is it possible to define the URI as `file.uri.split("?")[0]`?

Reply:
Got it, thanks for clarifying.

I'm thinking that we should update the file-based CDK to handle this situation. WDYT about putting your `get_file_uri` method into `*FileBasedCursor` (let's rename `get_file_uri` to `get_state_message_key` to be more general), and update `DefaultFileBasedCursor` to use that instead of `file.uri`? By default, its value will be `file.uri`, but you can overwrite it in your GCS `Cursor`.

Reply:
We can do this as a follow up CDK improvement after I will merge this PR. WDYT?

Reply:
Sure, that works for me.